### PR TITLE
Fix bounding box type definition

### PR DESCRIPTION
**Pull request by jeffbdye**
Addresses #49733.

`npm test @mapbox/mapbox-sdk` fails locally due to having named tuple members based on @techieshark, going to see if it's just something on my local machine or if it's enforced by the build too. Worst case, I'll fall back on just updating the type to be `[number, number, number, number]` to at least get past the truly incompatible types.

Somewhat related, I saw @SKalt in the mapbox-sdk-js repo was working on automatically generating types from the jsdoc type hints [here](https://github.com/mapbox/mapbox-sdk-js/pull/361) - I'll have to check in and see where the status of that is since it'd be preferable to have the types be more automatically maintained.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes). 
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/mapbox/mapbox-sdk-js/blob/f71b570b48d96f2aace58c6ad7cf3fb19c946e8e/services/geocoding.js#L100
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by jeffbdye** [comment]
Ok, seems like the labeled tuple elements require typescript 4.0 or later based on the build failure. Is typescript 4.0 fine to specify given that it was released around [August](https://devblogs.microsoft.com/typescript/announcing-typescript-4-0/), or should I fall back on non-labeled elements and make sure the use of `BoundingBox` is well documented? From a quick search, I'm not seeing many other definitions specifying 4.0 though I'm not sure if I was searching correctly.

**Comment by jeffbdye** [comment]
I tried setting `// Minimum Typescript Version: 4.0` but some errors came up with the geojson and mapbox-gl types that this depends on when tested locally. For now, I'm going to opt to fix the incorrect type and we can revisit bumping typescript versions to take advantage of the labeled tuples later unless there's a straightforward fix (which I might be missing, please let me know!).

File: types/mapbox__mapbox-sdk/index.d.ts [hunk 1]
```diff
@@ -1,1752 +1,1753 @@
...
+   *********************************************************************************************************************/
+
+  export default function Geocoding(config: SdkConfig | MapiClient): GeocodeService;
+
+  interface GeocodeService {
+    forwardGeocode(request: GeocodeRequest): MapiRequest;
+    reverseGeocode(request: GeocodeRequest): MapiRequest;
+  }
+
+  type BoundingBox = [
```
Comment (jeffbdye) [inline types/mapbox__mapbox-sdk/index.d.ts:813]: All the blank space changes aside, this is the type change in question for this PR.

Comment (techieshark) [reply types/mapbox__mapbox-sdk/index.d.ts:813]: 👍

**Review by techieshark — APPROVED** [review]
Not sure if an approving review from me helps, but the changes called out (list of four `number`s) look good to me

**Comment by jeffbdye** [comment]
Ready to merge

**Pull request merged**